Sources/Bow/Effects/Typeclasses/MonadDefer.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+
+public protocol MonadDefer : MonadError where E == Error {
+    func deferExecution<A>(_ fa : @escaping () -> Kind<F, A>) -> Kind<F, A>
+}
+
+public extension MonadDefer {
+    public func invoke<A>(_ f : @escaping () throws -> A) -> Kind<F, A> {
+        return self.deferExecution {
+            do {
+                return try self.pure(f())
+            } catch {
+                return self.raiseError(error)
+            }
+        }
+    }
+    
+    public func lazy() -> Kind<F, Unit> {
+        return invoke {}
+    }
+    
+    public func deferUnsafe<A>(_ f : @escaping () -> Either<Error, A>) -> Kind<F, A> {
```

Comment:
AFAIK this is not unsafe but pure because `suspend` and `delay` defer execution of any effects.

Reply:
I thought the same, but I used the same name as in Arrow. I've changed it to `delayEither` and we may think a better name later.